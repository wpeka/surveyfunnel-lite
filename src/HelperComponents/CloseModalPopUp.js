import { ModalContext } from '../Components/Context/ModalContext';
import { useContext } from "react";
import React from "react";
export function CloseModal(props) {
	let { setCurrentElement, setShowModal } = useContext(ModalContext);
	const closeModalFunction = () => {
		setCurrentElement(false);
		setShowModal(false);
	}
	return (
		<>
			<button className={props?.buttonClass ? props.buttonClass : ''} onClick={closeModalFunction}>{props?.closeText ? props.closeText : 'X'}</button>
		</>
	)
}