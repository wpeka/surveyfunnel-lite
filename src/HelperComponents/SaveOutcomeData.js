import React, { useContext } from 'react';
import {BuildContext} from '../Components/Context/BuildContext';
function SaveOutcomeData({getList}) {
    const {saveDataOutcome}=useContext(BuildContext);
    const saveData=()=>{
    const List=getList();
    saveDataOutcome(List);
    };
    return (
        <div className="modalComponentSaveButton">
            <button onClick={saveData}>Save Data</button>
        </div>
    );
}

export default SaveOutcomeData;