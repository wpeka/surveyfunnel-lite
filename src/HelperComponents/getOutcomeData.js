import React,{useContext} from 'react';
import {BuildContext} from '../Components/Context/BuildContext'

function getOutcomeData(props) {
    const {outComeData}=useContext(BuildContext);
    return outComeData;
}

export default getOutcomeData;