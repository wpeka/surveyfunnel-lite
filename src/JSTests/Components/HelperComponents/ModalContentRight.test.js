/**
 * @jest-environment jsdom
 */
import React from "react";
import { convertToRgbaCSS, designBackground } from "../../../HelperComponents/HelperFunctions";
import { useEffect } from "react";
import ModalContentRight from "../../../HelperComponents/ModalContentRight";
import renderer from 'react-test-renderer';


test('HelperFunctions convertToRgbaCSS test', () => {

    const designCon={
            fontFamilyValue:'200',
            fontColor:'blue',
            backgroundStyle:'none',
            fontFamily:'sans',
            backgroundContainerColor:'red',
            selectedImageUrl:'none'
        }
        
    const currentElement= 'content-element'


    const component1 = renderer.create( <ModalContentRight designCon={designCon} currentElement={currentElement} /> );
	let tree = component1.toJSON();
	expect(tree).toMatchSnapshot();
    console.log(tree);
});

it("When fontFamilyValue is not set then useEffect should return",()=>{
    const designCon={
        fontFamilyValue:'',
        fontColor:'blue',
        backgroundStyle:'none',
        fontFamily:'sans',
        backgroundContainerColor:'red',
        selectedImageUrl:'none'
    }
    
const currentElement= 'content-element'


const component1 = renderer.create( <ModalContentRight designCon={designCon} currentElement={currentElement} /> );
let tree = component1.toJSON();
expect(tree).toMatchSnapshot();
console.log(tree);
})