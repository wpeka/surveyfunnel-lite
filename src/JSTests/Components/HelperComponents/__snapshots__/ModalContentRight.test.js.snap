// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelperFunctions convertToRgbaCSS test 1`] = `
<div
  className="image-container"
  style={
    Object {
      "background": "linear-gradient(rgba(255,255,255,undefined), rgba(255,255,255,undefined)), url('none') 50%/cover",
      "color": "rgba(undefined, undefined, undefined, undefined)",
      "fontFamily": "sans",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <div
    className="preview"
  >
    <div
      className="main-tab-container"
    >
      <div
        className="tab-list surveyfunnel-lite-tab-content-element"
        style={
          Object {
            "background": "rgba(undefined, undefined, undefined, undefined)",
          }
        }
      >
        <div
          className="tab-container"
        >
          <div
            className="tab"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`When fontFamilyValue is not set then useEffect should return 1`] = `
<div
  className="image-container"
  style={
    Object {
      "background": "linear-gradient(rgba(255,255,255,undefined), rgba(255,255,255,undefined)), url('none') 50%/cover",
      "color": "rgba(undefined, undefined, undefined, undefined)",
      "fontFamily": "sans",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <div
    className="preview"
  >
    <div
      className="main-tab-container"
    >
      <div
        className="tab-list surveyfunnel-lite-tab-content-element"
        style={
          Object {
            "background": "rgba(undefined, undefined, undefined, undefined)",
          }
        }
      >
        <div
          className="tab-container"
        >
          <div
            className="tab"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
